File: aula 2/aula3/personagem.py
class Personagem:

    def __init__(self, nome, vida, nivel):
        self.nome = nome
        self.vida = vida
        self.nivel = nivel

    @property
    def nome(self):
        return self._nome

    @nome.setter
    def nome(self, nome):
        if nome != "":
            self._nome = nome
        else:
            print("Nome não pode ser vazio.")

    @property
    def vida(self):
        return self._vida

    @vida.setter
    def vida(self, vida):
        if vida >= 0 and vida <= 100:
            self._vida = vida
        else:
            print("Vida deve estar entre 0 e 100.")

    @property
    def nivel(self):
        return self._nivel

    @nivel.setter
    def nivel(self, nivel):
        if nivel >= 1:
            self._nivel = nivel
        else:
            print("Nivel deve ser maior ou igual a 1.")

    def ficha(self):
        return "Nome: " + self.nome + "\nVida: " + str(self.vida) + "\nNivel: " + str(self.nivel)

print("-----Personagem-----")
heroi = Personagem("Jane", 100, 1)
print(heroi.ficha())

print("\n-----Testando valores validos-----")
heroi.nome = "Angela"
heroi.vida = 80
heroi.nivel = 5
print(heroi.ficha())

print("\n-----Testando valores invalidos-----")
heroi.nome = ""
heroi.vida = 150
heroi.vida = -10
heroi.nivel = 0

print("\n-----Valores finais-----")
print(heroi.ficha())

File: aula 2/aula3/Personagem.java
class Personagem {

    private String nome;
    private int vida;
    private int nivel;

    public Personagem(String nomePersonagem, int vidaPersonagem, int nivelPersonagem) {
        setNome(nomePersonagem);
        setVida(vidaPersonagem);
        setNivel(nivelPersonagem);
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        if (!nome.isEmpty()) {
            this.nome = nome;
        } else {
            System.out.println("Nome não pode ser vazio.");
        }
    }

    public int getVida() {
        return vida;
    }
    public void setVida(int vida) {
        if (vida >= 0 && vida <= 100) {
            this.vida = vida;
        } else {
            System.out.println("Vida deve estar entre 0 e 100.");
        }
    }

    public int getNivel() {
        return nivel;
    }

    public void setNivel(int nivel) {
        if (nivel >= 1) {
            this.nivel = nivel;
        } else {
            System.out.println("Nivel deve ser maior ou igual a 1.");
        }
    }

    String ficha() {
        return "Nome: " + nome + "\nVida: " + vida + "\nNivel: " + nivel;
    }
}
public class Main {

    public static void main(String[] args) {

        Personagem heroi = new Personagem("Jane", 100, 1);

        System.out.println("-----Personagem-----");
        System.out.println(heroi.ficha());

        System.out.println("\n-----Testando valores validos-----");

        heroi.setNome("Angela");
        heroi.setVida(80);
        heroi.setNivel(5);

        System.out.println(heroi.ficha());

        System.out.println("\n-----Testando valores invalidos-----");

        heroi.setNome("");
        heroi.setVida(150);
        heroi.setVida(-10);
        heroi.setNivel(0);

        System.out.println("\n-----Valores finais-----");
        System.out.println(heroi.ficha());
    }
}
